ReactCommon/react/renderer/attributedstring/AttributedString.cpp

```diff
@@ -136,7 +136,7 @@ bool AttributedString::isContentEqual(const AttributedString &rhs) const {
     return false;
   }
 
-  for (auto i = 0; i < fragments_.size(); i++) {
+  for (unsigned i = 0; i < fragments_.size(); i++) {
```

Comment:
Is this supposed to be `size_t`?

Reply:
mmm good question! I was following the format of this file and they seemed to use unsigned i when comparing fragments_.size(). Should I change both of them to size_t?
 (https://github.com/facebook/react-native/blob/0cc40a1de136dce1fff854876cbd19893026c4d3/ReactCommon/react/renderer/attributedstring/AttributedString.cpp#L116)

Reply:
`fragments_` is of a type `butter::vector`, which is conditionally either s `std::vector`, or `folly::small_vector`. Both of the latter return `size_t` for `size()`, so it would be the most correct here. Replacing the existing usage would be appreciated.

`size_t` and `unsigned` are both 32 bit unsigned integers on 32 bit executables, but `size_t` is 64 bits when targeting 64 bit systems. So the different types here are:
1. `auto`: Inferred from 0 to be a 32 bit signed integer
2. `unsigned`: 32 bit unsigned integer
3. `size_t`: Unsigned type of variable width used for vector size

In practice the world would blow up much earlier if we ever had even 2^31 nested text elements, but only `size_t` would fully prevent truncation here.

We should also generally prefer stdint types like `uint32_t` for these fixed width types instead of `unsigned`. I don't know the history of how it was added, but it looks like it was there for a while.

Reply:
thanks for explanation, I'm new to C so it's super helpful to have people explain the various nuances!

I've switched both to size_t

Reply:
This is a great reference: https://en.cppreference.com/w/cpp/language/types

`cppreference` is dense, but it's a great resource for trying to remember its nuances. I needed to use it to remind myself of the guarantees for `unsigned int` 😅.